WalletWasabi/Logging/Logger.cs

```diff
@@ -219,12 +219,15 @@ public static void Log(LogLevel level, string message, int additionalEntrySepara
 
 				IoHelpers.EnsureContainingDirectoryExists(FilePath);
 
-				if (File.Exists(FilePath))
+				if (MaximumLogFileSize > 0)
```

Comment:
The changes in `Logger.cs` should probably be a standalone PR or should be reverted. But personally, it seems very useful to me to have these changes because any arbitrary number we put here will lead to log trimming and then the logs are mostly worthless to do some serious debugging. 

Or if I put it in an other way, if the `Logger.cs` changes were in place sooner, I would spend about several days less time on this PR in total.

Reply:
> For example, I was not able to use 100Mb Log file with Notepad++ because it was lagging and I have a pretty fast computer.

This is indeed an issue but it seems that Notepad++ is [not a good editor for big files (old thread though)](https://superuser.com/questions/286106/how-to-open-a-big-file-in-notepad) (as it's not the goal of its authors and it does not matter how fast your computer is because support for big files typically require for an editor to use different text algorithms for highlighting and other stuff). There are not that many good options, I personally use vscode and it seems to work good with big files too. 

> You have to put some effort to reduce the rate of the trace messages generated by your code - that is the solution that will scale.

So I understand what you are saying but let me say a few notes about why these detailed log messages may be useful:

* There is certain class of bugs that are very hard to discover using unit tests / regression tests and these are various race conditions, discovering situations "this can never happen", networking is notoriously hard etc. If one has these trace logs, the chance of getting to the bottom of these issues is much higher.  In my experience, when one does not have logs, then issues that appear rarely are almost never fixed (because they are hard/impossible to reproduce).
* 100 MB log seems high but then, five years ago, I worked on a software package that generated like gigabytes of logs per day. Basically, if you have logs, then you know, if you don't then you guess. 
    * Yes, some editors can't handle log files that big but then the goal is to be able to diagnose software run and find bugs fast(er), so I just use an editor that can handle that.
* I tend to think that I'm the only user of these trace level log lines. Does it affect others then?

However, I think there are really some log lines that can be removed, I can have a look but in general, I think we would much more benefit from having a command line option `--LogCategory=net` which would lead to logging of log messages that have this `net` category (the idea is from Bitcoin Core which this concept of log categories).

I would be unhappy to remove a lot of trace log messages because for each Tor issue, I would need to re-add them and that seems like a waste of time really. And this touches the issue whether you are opposed to having that many trace log level log lines in code itself or not.

Reply:
TRACE is good as the final level - it will be only used by devs.